### PR TITLE
do not regather ice candidates when renegotiating a bundle connection

### PR DESCRIPTION
**Pull request by koush**

### CODE REVIEW

**Comment by koush** [comment]
Fixes https://github.com/shinyoshiaki/werift-webrtc/issues/269

**Comment by koush** [comment]
This fixes the spurious socket allocation. Problems that remain that could be in a separate change:

SDP contains redundant candidates (Chrome does not resend them on renegotiation)
non bundle connections still generate a ton of candidates that are never cleaned up (but I'm not inclined to fix this, as it's a minor issue)

File: packages/webrtc/src/peerConnection.ts [hunk 1]
```diff
@@ -634,9 +634,19 @@ export class RTCPeerConnection extends EventTarget {
...
 
     // # gather candidates
-    await Promise.all(
-      this.iceTransports.map((iceTransport) => iceTransport.iceGather.gather())
+    const connected = this.iceTransports.find(
+      (transport) => transport.state === "connected"
     );
+    if (this.remoteIsBundled && connected) {
+      // no need to gather ice candidates on an existing bundled connection
+      connected.iceGather.gather();
```
Comment (shinyoshiaki) [inline packages/webrtc/src/peerConnection.ts:642]: What is the intent of "gather" without "await" here?

Comment (koush) [reply packages/webrtc/src/peerConnection.ts:642]: This is a bug, I will fix this. Thanks!

**Pull request merged**